envs in overrides was not applied

<!--
    ESLint adheres to the [JS Foundation Code of Conduct](https://js.foundation/community/code-of-conduct).

    This template is for bug reports. If you are here for another reason, please see below:

    1. To propose a new rule: https://eslint.org/docs/developer-guide/contributing/new-rules
    2. To request a rule change: https://eslint.org/docs/developer-guide/contributing/rule-changes
    3. To request a change that is not a bug fix, rule change, or new rule: https://eslint.org/docs/developer-guide/contributing/changes
    4. If you have any questions, please stop by our chatroom: https://gitter.im/eslint/eslint

    Note that leaving sections blank will make it difficult for us to troubleshoot and we may have to close the issue.
-->

**Tell us about your environment**

* **ESLint Version:** v5.16.0
* **Node Version:** v10.15.3
* **npm Version:** 6.9.0

**What parser (default, Babel-ESLint, etc.) are you using?**

**Please show your full configuration:**

<details>
<summary>Configuration</summary>

<!-- Paste your configuration below: -->
```js
  "eslintConfig": {
    "extends": [
      "eslint:recommended"
    ],
    "overrides": [
      {
        "files": [
          "./lib/*.js"
        ],
        "env": {
          "browser": true
        }
      }
    ]
  },
```

</details>

**What did you do? Please include the actual source code causing the issue, as well as the command that you used to run ESLint.**

<!-- Paste the source code below: -->
```js
'use strict';

window.console.log('hello');
```

<!-- Paste the command you used to run ESLint: -->
```bash

```

**What did you expect to happen?**
not error reported, since it has setting `env "browser": true`

**What actually happened? Please include the actual, raw output from ESLint.**
```bash
/Users/weiran/repo/work/eslint-issue/lib/index.js
  3:1  error  'window' is not defined  no-unde
```

**Are you willing to submit a pull request to fix this bug?**
yes

--- a/lib/cli-engine/config-array/override-tester.js
+++ b/lib/cli-engine/config-array/override-tester.js
@@ -55,7 +55,17 @@ function toMatcher(patterns) {
     if (patterns.length === 0) {
         return null;
     }
-    return patterns.map(pattern => new Minimatch(pattern, minimatchOpts));
+    return patterns.map(pattern => {
+        if (/^\.[/\\]/u.test(pattern)) {
+            return new Minimatch(
+                pattern.slice(2),
+
+                // `./*.js` should not match with `subdir/foo.js`
+                { ...minimatchOpts, matchBase: false }
+            );
+        }
+        return new Minimatch(pattern, minimatchOpts);
+    });
 }
 
 /**

--- a/tests/lib/cli-engine/config-array/override-tester.js
+++ b/tests/lib/cli-engine/config-array/override-tester.js
@@ -191,10 +191,10 @@ describe("OverrideTester", () => {
         match("foo.js", ["*.js"], []);
         match("foo.js", ["**/*.js"], []);
         match("bar.js", ["*.js"], ["foo.js"]);
+        match("foo.js", ["./foo.js"], []);
+        match("foo.js", ["./*"], []);
+        match("foo.js", ["./**"], []);
 
-        noMatch("foo.js", ["./foo.js"], []);
-        noMatch("foo.js", ["./*"], []);
-        noMatch("foo.js", ["./**"], []);
         noMatch("foo.js", ["*"], ["foo.js"]);
         noMatch("foo.js", ["*.js"], ["foo.js"]);
         noMatch("foo.js", ["**/*.js"], ["foo.js"]);
@@ -208,11 +208,11 @@ describe("OverrideTester", () => {
         match("subdir/foo.js", ["subdir/foo.js"], []);
         match("subdir/foo.js", ["subdir/*"], []);
         match("subdir/second/foo.js", ["subdir/**"], []);
+        match("subdir/foo.js", ["./**"], []);
+        match("subdir/foo.js", ["./subdir/**"], []);
+        match("subdir/foo.js", ["./subdir/*"], []);
 
         noMatch("subdir/foo.js", ["./foo.js"], []);
-        noMatch("subdir/foo.js", ["./**"], []);
-        noMatch("subdir/foo.js", ["./subdir/**"], []);
-        noMatch("subdir/foo.js", ["./subdir/*"], []);
         noMatch("subdir/foo.js", ["*"], ["subdir/**"]);
         noMatch("subdir/very/deep/foo.js", ["*.js"], ["subdir/**"]);
         noMatch("subdir/second/foo.js", ["subdir/*"], []);

